packages/aws-cdk-lib/aws-appsync/lib/appsync-function.ts

```diff
@@ -161,7 +161,8 @@ export class AppsyncFunction extends Resource implements IAppsyncFunction {
       runtime: props.runtime?.toProperties(),
       codeS3Location: code?.s3Location,
       code: code?.inlineCode,
-      functionVersion: '2018-05-29',
+      // function Version should not be set when Code is provided
```

Comment:
I'll give this a more thorough look on Monday, but could explain in this commend why the version should not be set? Does this publish a new version implicitly and we wind up using `$LATEST` if the version is `undefined` here?

Reply:
`functionVersion` is a setting only used when the function runtime is `VTL` (the AppSync default), when the runtime is set to `APPSYNC_JS` and `code` is specified, `functionVersion` should not be used (it is ignored).

Reply:
Please add that information to the comment on line 164, to make it clear why this is the way it is

Reply:
addressed in latest commit